src/index.ts

```diff
@@ -27,30 +51,44 @@ export function _buildMetadataAccessor(type: string) {
       property = '/' + options.property;
     }
 
-    const reqOpts = extend(
-                        true, {
-                          uri: BASE_URL + '/' + type + property,
-                          headers: {'Metadata-Flavor': 'Google'}
-                        },
-                        options) as _request.Options &
-        {property?: string};
+    try {
+      validate(options);
+    } catch (e) {
+      if (callback) {
```

Comment:
I read an assertion in a comment below that `callback` cannot by null here. If that is really the case, then all these `if (callback)` guards can be removed and the rethrow is not necessary.

Given that this API is callback based, it would be better to avoid a throw in `validate` so that you can the try-catch here becomes unnecessary. Consistency of style.

Reply:
Removed all that jazz and made validate callback based